xapian-core/weight/pl2plusweight.cc

```diff
@@ -0,0 +1,153 @@
+/** @file pl2plusweight.cc
+ * @brief Xapian::PL2PlusWeight class - the PL2+ weighting scheme of the DFR framework.
+ */
+/* Copyright (C) 2016 Vivek Pal
```

Comment:
Again, you should preserve the copyright statements from the file you based this on.

You should also probably add copyright statements for your own contributions to files which you've made copyrightable changes to.  The dividing line of what's copyrightable is surprisingly complex but for example you've clearly made sufficient contributions to `include/xapian/weight.h`.